include/net/ieee802154_radio.h

```diff
@@ -24,6 +24,14 @@ extern "C" {
  * @{
  */
 
+enum ieee802154_hw_caps {
+	IEEE802154_HW_FCS		= BIT(0),
+	IEEE802154_HW_PROMISC		= BIT(1),
+	IEEE802154_HW_FILTER		= BIT(2),
+	IEEE802154_HW_CSMA		= BIT(3),
+	IEEE802154_HW_2_4_GHZ		= BIT(4),
```

Comment:
What do you think about:
```
IEEE802154_HW_ BPSK = BIT(4),
IEEE802154_HW_O-QPSK =BIT(5),
IEEE802154_HW_2450_DSSS = BIT(6),
IEEE802154_HW_UWB_LB = BIT(7), /* USB low band */
```
?

Reply:
Hum, not sure I'll add those. I went for the most generic one first, those affecting directly the L2 layer and the drivers we already have.

About the *PSK, DSSS and UWB (Ultra-Wide Band btw, it has nothing to do with USB ;) ), I am not convinced L2 will ever touch things about modulation. For instance, about cc1200, switching from one to the other requires a lot of parts, fully hw centric, no really possible to make generic. So basically, it is then up to the user to provide the RF settings (a default one and more if he wants to) and select the one he wants, feeding the driver with it.

Reply:
I thought rather in the direction to verify which channel is valid.
(It would be more correct to have:
IEEE802154_HW_UWB_BG0 and IEEE802154_HW_UWB_BG1)
So for UWB_BG0 the valid channels are: 0
for UWB_BG1 the valid channels are: 1 .. 15

Reply:
No need for this, it's addressed in further patches and again, there are so many bands varieties that the L2 won't keep any knowledge of these, it would just eat up ROM (13 carrier bands, on which are applied from 1 to sometime up to 7 different modulations, all of these with different channel limits... it's a mess!). Instead, the drivers will have to provide the supported number of channels, according to the RF settings it currently has installed.